packages/react-components/src/components/Switch/Switch.tsx

```diff
@@ -110,6 +110,7 @@ export const Switch: React.FC<SwitchProps> = ({
           )}
           {isLocked && (
             <Icon
+              customColor={styles[`${baseClass}__icon`]}
```

Comment:
I think it would be better to just apply a className here eg,  `className={styles[`${baseClass}__icon`]}`, and set up a color there. It will be easier due to the fact, that in order to use CSS var in JavaScript it is required to compute it -> https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties#values_in_javascript.

I believe the current solution does not work as `customColor` expects a valid color value, while you provide a className with CSS styles.